src/reactive/keepalived.py

```diff
@@ -103,3 +107,17 @@ def loadbalancer_available(loadbalancer):
 @hook('upgrade-charm')
 def upgrade_charm():
     clear_flag('keepalived.started')
+
+
+@hook('pre-series-upgrade')
+def pre_series_upgrade():
+    service_pause('keepalived')
+    service_pause('procps')
+    status_set('blocked', 'Series upgrade in progress')
```

Comment:
Seems like this status could easily get overwritten by other handlers. I recommend setting it in an atexit handler or refactoring the charm to use layer-status.